calico_kubernetes/calico_kubernetes.py

```diff
@@ -181,32 +227,43 @@ def _container_add(self, pid, ip, interface, hostname, orchestrator_id):
         # Obtain information from Docker Client and validate container state
         self._validate_container_state(self.docker_id)
 
-        # Assign ip address through IPAM Client
-        try:
-            ip_assigned = self._datastore_client.assign_address(None, ip)
-        except PoolNotFound:
-            logger.error("IP address %s does not belong to any configured "
-                         "pools. Exiting.", ip)
-            sys.exit(1)
+        if CALICO_IPAM == 'calico':
```

Comment:
This whole bit could be commonized into a method called "assign_container_ip" which is called by container add - simplifies the container_add code a bit.